truffleruby/src/main/java/org/truffleruby/core/regexp/RegexpNodes.java

```diff
@@ -749,30 +618,11 @@ private static Frame getMatchDataFrameSearchingDeclarations(Frame topFrame, bool
         return frame;
     }
 
-    private static FrameSlot getMatchDataFrameSlotWrite(MaterializedFrame frame) {
-        FrameDescriptor fd = frame.getFrameDescriptor();
-        FrameSlot fs = fd.findFrameSlot("$~");
-        if (fs == null) {
-            fs = fd.addFrameSlot("$~", FrameSlotKind.Object);
-        }
-        return fs;
-    }
-
-
     @TruffleBoundary
     private static FrameSlot getMatchDataSlot(Frame frame) {
         return frame.getFrameDescriptor().findFrameSlot("$~");
     }
 
-    private static ThreadLocalObject getMatchDataThreadLocalSearchingDeclarations(
-            RubyContext context,
-            Frame topFrame,
-            boolean add) {
-
-        final Frame frame = getMatchDataFrameSearchingDeclarations(topFrame, add);
-        return getThreadLocalObject(context, frame, add);
-    }
-
     @TruffleBoundary
     public static ThreadLocalObject getMatchDataThreadLocalSearchingStack(RubyContext context) {
```

Comment:
The other `getMatchData*` methods can probably also go away if `rb_backref_get` gets refactored into something simpler and based on get some (n) caller frame + ThreadLocalInFrameNode. But good to leave this for another PR.